api/v3/openapi.yaml

```diff
@@ -1658,6 +1815,76 @@ components:
                   - name: 'compression.type'
                     value: 'gzip'
 
+    ProduceRequest:
+      description: 'A single record to be produced to Kafka. To produce multiple records on the same
+                    connection, simply concatenate all the records, e.g.:
+                    {"partition_id":1}{"partition_id":2}. Delivery reports will be concatenated on
+                    the same order as the records are sent. See examples for the options available.'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ProduceRequest'
+          examples:
+            binary_and_avro_with_subject_and_raw_schema:
+              description: 'If using type: "BINARY" or type: "JSON", or using schema, type is
+                            required.'
```

Comment:
This doesn't read very well to me.
AFAIU if the key or value content uses one of the types where the serde procedure is known beforehand - binary or JSON - then we need `type` to designate that we don't need a schema lookup to deserialize.
If the key or value content uses one of the schema-serialized types, like Avro or Protobuf, I'm not sure why the type is needed too - isn't the schema reference enough?
Also _or using schema_ seems to be a bit at odds with _If not setting type, the record is assumed to use a schema_ from the next example.
There's probably a very good explanation behind all of this, but I don't think it's very obvious from the descriptions here and below.

Reply:
By "schema", I mean "the schema field". You when you the raw schema (so that Kafka REST register the schema for you), you need to say what type that schema is. We can't guess from the raw schema string (we probably could, but we don't).

Reply:
Changed from `schema` to `the schema field`.